### DISCUSSION TITLE
v3.0

### DISCUSSION THREAD

**Discussion by osy** [osy86/HaC-Mini#575] [Releases]
**IMPORTANT READ THE FOLLOWING BEFORE UPDATING**: The new OpenCore has an incompatible change with v2.17 and below. As a result, if you run the update and restart, you will not be able to find the boot device. There are a few ways to complete the update:

* Option 1: install v3.0 on a USB drive with the "Show boot menu" option set. Then boot into the USB drive which loads OpenCore and select the Reset NVRAM option. Then you must boot from the USB again and select macOS and install v3.0 on your main partition.
* Option 2: install v3.0 on your main partition and *also* on a USB drive (with “Show boot menu” set). Reboot and select your USB drive. Select Reset NVRAM and remove the USB drive when it reboots.
* Option 3: if you've already installed v3.0 on your main partition without doing the above, you can get into UEFI Shell and type in `fs0:\EFI\OC\OpenCore.efi` from the shell prompt.
* Option 4: if all else fails, you can do a BIOS reset which should reset all settings as well as the boot options. Make sure to set up the BIOS settings again with your old settings!

## Changes

* Add fans, voltage, and temperature sensors (#553)
* Translate to German (thanks @SkyrilHD)
* Intel wireless option now installs AirportItlwm instead of Itlwm

## Driver Updates

* AppleALC 1.5.6 => 1.6.0
* BrcmPatchRAM 2.5.5 => 2.5.8
* IntelMausi 1.0.5 => 1.0.6
* Lilu 1.5.0 => 1.5.3
* NVMeFix 1.0.5 => 1.0.7
* OpenCore 0.6.5 => 0.6.9
* VirtualSMC 1.1.9 => 1.2.3
* WhateverGreen 1.4.6 => 1.4.9

## Release Schedule

As this project reaches maturity, HaC mini will only be updated for bug fixes and major features. We will no longer be updating the driver KEXTs monthly.

We aim to have two major releases a year: one targeting a major update for macOS due to anticipated major changes (for example when macOS 12 comes out). A second one mid-way through the update cycle. Each major update will increment the second version number (for example v3.0 => v3.1). A major update will include any new features as well as bumping all drivers to the latest released versions.

When bug fixes are needed, we will only update the relevant drivers and the third version number will be incremented (for example v3.0 => v3.0.1).

Each major update will first be released as a pre-release as marked on Github. If some time passes and no major bugs are reported, then the pre-release will be promoted to the full release and will be the recommended build to install. Non-advanced users are recommended to only install full release builds as they will be more stable.

When reporting an issue, please make sure to include the version you are using.

<hr /><em>This discussion was created from the release <a href='https://github.com/osy/HaC-Mini/releases/tag/v3.0'>v3.0</a>.</em>

**Comment by KrisCris** [comment]
I followed option 1, while still end up having no bootable drive detected.

**Reply by KrisCris** [reply]
![13284491-573F-4FCC-BB84-DD2F356A7E49](https://user-images.githubusercontent.com/38860226/116989512-6facc480-ad04-11eb-8374-9a61d7d06217.jpeg)

Can you please provide an simple guide on how to find and load the efi? I have no idea how to get into any of these 4 drives listed in the image. (I tried command like cd but no luck)

**Reply by goodbest** [reply]
Instead of the command line with opencore uefi shell, 
Maybe you can enter the BIOS settings,  and load the `EFI/OC/OpenCore.efi` boot entry.

**Reply by KrisCris** [reply]
> Instead of the command line with opencore uefi shell,
> Maybe you can enter the BIOS settings, and load the `EFI/OC/OpenCore.efi` boot entry.

By double clicking this? 
![2347876C-BDD4-42A1-9F57-BDFBFC12A915](https://user-images.githubusercontent.com/38860226/116992628-ecda3880-ad08-11eb-85eb-0cae7dc1007b.jpeg)

Sadly my nuc still says a bootable device has not been detected.

**Reply by KrisCris** [reply]
Nvm, I reset the entire bios and it works

**Reply by osy** [reply]
I’m too late but you have to make sure not to boot the old OpenCore again after hitting reset NVRAM. (I.e. make sure to boot from the USB). I’ve clarified this in the instructions.

**Reply by KrisCris** [reply]
> I’m too late but you have to make sure not to boot the old OpenCore again after hitting reset NVRAM. (I.e. make sure to boot from the USB). I’ve clarified this in the instructions.

Yeah I know we need to boot from the new OpenCore. Perhaps I selected the wrong bootable drive in the boot menu, idk.
Anyway, thanks for the reply:)

**Comment by Needham73** [comment]
After update to V3.0 the computer stoped booting with the following screen:

![1](https://user-images.githubusercontent.com/83665449/117046599-c4ead500-ad08-11eb-9b67-55fb96df21c9.jpeg)

**Reply by osy** [reply]
Can you check enable debug and reset boot-args in the installer and get a picture of the panic with more info?

**Reply by Needham73** [reply]
With those options I got the same screen

![2](https://user-images.githubusercontent.com/83665449/117067987-9c6fd480-ad22-11eb-8ac4-f074c110c6c1.jpeg)

**Reply by osy** [reply]
What install options did you choose? Can you dump your EFI directory and attach it?

**Reply by Needham73** [reply]
![Screen Shot 2021-05-04 at 21 59 25](https://user-images.githubusercontent.com/83665449/117069185-1eacc880-ad24-11eb-925c-7f5a831018e2.png)
![Screen Shot 2021-05-04 at 21 59 52](https://user-images.githubusercontent.com/83665449/117069197-22404f80-ad24-11eb-86ad-43d3fa07c56d.png)
![Screen Shot 2021-05-04 at 22 00 10](https://user-images.githubusercontent.com/83665449/117069214-24a2a980-ad24-11eb-86a1-81d784dd1f01.png)

**Reply by Needham73** [reply]
Here is the list of all the files in EFI partition:

[efi.txt](https://github.com/osy/HaC-Mini/files/6423935/efi.txt)

**Reply by osy** [reply]
Can you attach OC's config.plist? Also I think the issue is with the new Intel Wireless drivers. Try disabling that (after giving me config.plist)

**Reply by Needham73** [reply]
Yes. If I remove the Intel Wireless drivers it boots.


[config.plist.zip](https://github.com/osy/HaC-Mini/files/6423986/config.plist.zip)

**Reply by Needham73** [reply]
I installed it manually and its working now.

			<dict>
				<key>BundlePath</key>
				<string>AirportItlwm.kext</string>
				<key>Enabled</key>
				<true/>
				<key>ExecutablePath</key>
				<string>Contents/MacOS/AirportItlwm</string>
				<key>PlistPath</key>
				<string>Contents/Info.plist</string>
			</dict>

**Reply by osy** [reply]
Can you try https://github.com/osy/HaC-Mini/suites/2651808749/artifacts/58295515 and see if the installer is fixed?

**Reply by osy** [reply]
Also let me know if AirportItlwm works and if it's better than Itlwm before.

**Reply by Needham73** [reply]
The installer is fixed and AirportItlwm is working
